modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -0,0 +1,397 @@
+  #Generate parameter needs to be run before this to generate the samples. This is hopefully done in the main workflow.
+  load(file.path(settings$outdir, "samples.Rdata"))  ## loads ensemble.samples
+  #reformatting params
+  params <- list()
+  for (i in seq_len(nens)) {
+    params[[i]] <- lapply(ensemble.samples, function(x, n) {
+      x[i, ] }, n = i)
+  } 
+  new.params <- params
+
+  ###-------------------------------------------------------------------###
+  ### If this is a restart - Picking up were we left last time          ###
+  ###-------------------------------------------------------------------###----   
+  if (restart){
+    load(file.path(settings$outdir,"SDA", "sda.output.Rdata"))
+    #--- Updating the nt and etc
+    if(!dir.exists(file.path(settings$outdir,"SDA",assimyears[t]))) dir.create(file.path(settings$outdir,"SDA",assimyears[t]))
+    # finding/moving files to it's end year dir
+    files.last.sda<-list.files.nodir(file.path(settings$outdir,"SDA"))
+    #copying
+       file.copy(file.path(file.path(settings$outdir,"SDA"),files.last.sda),
+                 file.path(file.path(settings$outdir,"SDA"),paste0(assimyears[t],"/",files.last.sda))
+                 )
+       params<-new.params
+  }else{
+    t<-0
+  }
+
+  ###------------------------------------------------------------------------------------------------###
+  ### loop over time                                                                                 ###
+  ###------------------------------------------------------------------------------------------------###---- 
+  while(t<nt){
+    t<-t+1
+    # do we have obs for this time - what year is it ?
+    obs <- which(!is.na(obs.mean[[t]]))
+    obs.year <- year(names(obs.mean)[t])
+    ###-------------------------------------------------------------------------###
+    ###  Taking care of Forecast. Splitting /  Writting / running / reading back###
+    ###-------------------------------------------------------------------------###-----  
+    #- Check to see if this is the first run or not and what inputs needs to be sent to write.ensemble configs
+    if (t>1){
+      #removing old simulations
+      unlink(list.files(outdir,"*.nc",recursive = T,full.names = T))
+      #-Splitting the input for the models that they don't care about the start and end time of simulations and they run as long as their met file.
+      inputs.split <- list()
+      for(i in seq_len(nens)){
+        if(!no_split){ 
+     #---------------- model specific split inputs
+          inputs.split$samples[i] <-do.call(my.split_inputs, 
+                                      args = list(settings = settings, 
+                                                  start.time = (ymd_hms(obs.times[t-1],truncated = 3) + second(hms("00:00:01"))), 
+                                                  stop.time = obs.times[t],
+                                                  inputs = inputs$samples[[i]])) 
```

Comment:
where does `inputs` come from now?

Reply:
input is defined in line 232. The first round the met is split just to the bounds of SDA so I read it from the original settings but in the second round on I don't change the original met path and I just produce and use the split ones in each iteration.